### ISSUE TITLE
[v1.0.0] Passing CP Code ID to akamai_property results in crash

### ISSUE THREAD

**Issue by carnei-ro** [akamai/terraform-provider-akamai#198]
with provider 0.11.0 I could create a CP Code and then use the id as variable for `templatefile` function to my `akamai_property` rules.

```
resource "akamai_cp_code" "this" {
  name        = var.cp_code_name
  contract = var.contract_id
  group    = data.akamai_group.this.id
  # TODO: Change product to product_id when https://github.com/akamai/terraform-provider-akamai/pull/197 be merged
  product = var.product_id
}

resource "akamai_property" "this" {
  name        = var.property_name
  contract = var.contract_id
  group    = data.akamai_group.this.id
  product  = var.product_id
  contact = ["leandro.carneiro@ifood.com.br"]
  hostnames = {
    for host in var.hostnames :
    host => var.edge_hostname
  }
  rule_format = var.rule_format
  rules = templatefile("${path.module}/templates/rules.tpl", {
    behaviors = templatefile("${path.module}/templates/behaviorsDefault.tpl", {
      cp_code_id            = element(split("_", akamai_cp_code.this.id), 1),
      origin_hostname       = local.origin_hostname,
      sure_route_object_url = var.sure_route_object_url,
      cache_nostore         = local.cache_nostore,
    }),
    children  = jsonencode(var.children_properties),
    name      = "default",
    variables = jsonencode(var.property_variables)
  })

  depends_on = [ akamai_cp_code.this ]
}
```

**Comment by carnei-ro** [comment]
Ok, reading the `crash.log` and doing some experimentations the root cause is at `StateFunc` of `rules` in the schema map. It tries to `compactJSON` even then the string is not a Json yet. I'm opening a Pull Request with the validation.

**Comment by carnei-ro** [comment]
https://github.com/akamai/terraform-provider-akamai/pull/199

**Comment by lkowalsk-akamai-com** [comment]
This issue should be resolved in v1.2.1 release, which is available now.
